### ISSUE TITLE
Datadog RUM not publishing

### ISSUE THREAD

**Issue by bruno-watt-wcq** [backstage/backstage#25190]
### 📜 Description

Following the instructions we have added the datadog rum input into the configuration files.

Firstly, this is unclear in the instructions

```
f your app-config.yaml file does not have this configuration, you may have to adjust your packages/app/public/index.html to include the Datadog RUM init() section manually.
```

We presume this means if you dont want to use the app-config approach.

Data never arrives at Datadog, in the front end console we also see the warning that

```
Datadog Browser SDK: Session Replay Sample Rate should be a number between 0 and 100
```

Which tells me the module probably does exist in the front end. 

Weve set the config as such:

```
  datadogRum:
    clientToken: TOKEN
    applicationId: ID
    site: datadoghq.eu
    env: 'devops-green'
    sessionSampleRate: 100
    sessionReplaySampleRate: 1
```

No matter what we set the sample rate to that error is present.

We worked out that we needed to add this to the config:

```
script-src: ["'self'", "'unsafe-eval'", "'unsafe-inline'", 'https://www.datadoghq-browser-agent.com/eu1/v5/datadog-rum.js']
```

We still are not seeing any traffic getting to DD. Is there another manual step thats required to make this work?

Note: prior to setting up RUM we did a fresh build of EVERYTHING on version 1.27, getting rid of all the old backend cruft etc, so we are pure 1.27 now.

### 👍 Expected behavior

Populate the DD Rum application.

### 👎 Actual Behavior with Screenshots

Nothing. tho we do see this error in the console:

<img width="492" alt="Screenshot 2024-06-13 at 12 22 21 PM" src="https://github.com/backstage/backstage/assets/135942633/70d9859b-bdd1-4346-bc84-c4d55ed97e88">

### 👟 Reproduction steps

- configure RUM via the config.
- launch

### 🖥️ Your Environment

desktop (tilt-rancher)
AWS (EKS)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

No, I don't have time to work on this right now

**Comment by camilaibs** [comment]
Hi @bruno-watt-wcq 👋🏻,
Could you please copy and paste this section to your `packages/app/public/index.html` file and tell us if that solves the issue?
https://github.com/backstage/backstage/blob/27c3700433e0995bb9562c63f932ca1b75fa427e/packages/app/public/index.html#L46-L83

**Comment by darylgraham** [comment]
There is also an option to install it via npm directly into the frontend package: https://docs.datadoghq.com/real_user_monitoring/browser/setup/#npm

I could get it working with hardcoded values, but when I tried to switch over to getting the values from config via configApi it looked like I was trying to inject too early in the process and configApi wasn't available yet. If anyone knows how/where to add this, that would be great.

**Comment by secustor** [comment]
> There is also an option to install it via npm directly into the frontend package: https://docs.datadoghq.com/real_user_monitoring/browser/setup/#npm
> 
> I could get it working with hardcoded values, but when I tried to switch over to getting the values from config via configApi it looked like I was trying to inject too early in the process and configApi wasn't available yet. If anyone knows how/where to add this, that would be great.

This is the use case which lead me to create this issue https://github.com/backstage/backstage/issues/24561

**Comment by darylgraham** [comment]
> > There is also an option to install it via npm directly into the frontend package: https://docs.datadoghq.com/real_user_monitoring/browser/setup/#npm
> > I could get it working with hardcoded values, but when I tried to switch over to getting the values from config via configApi it looked like I was trying to inject too early in the process and configApi wasn't available yet. If anyone knows how/where to add this, that would be great.
> 
> This is the use case which lead me to create this issue #24561

I'm not deep into our backstage journey yet so happy to defer to those with more experience on this. It feels like this type of config would be ok to be loaded into the frontend when it's being built/packaged since it doesn't change that frequently.

**Comment by secustor** [comment]
If you want use environment tags in DD you have to build independently for each environment, as you said it its only build once while creating the Docker image. That said your case could be different if a more specialised setup.

**Comment by bruno-watt-wcq** [comment]
> Hi @bruno-watt-wcq 👋🏻, Could you please copy and paste this section to your `packages/app/public/index.html` file and tell us if that solves the issue?
> 
> https://github.com/backstage/backstage/blob/27c3700433e0995bb9562c63f932ca1b75fa427e/packages/app/public/index.html#L46-L83

Yep this was the second thing we tried. with both versions of the plugin, both with the same outcome, nothing but the error listed originally.

**Comment by bruno-watt-wcq** [comment]
Correction on the above. I think when we said we saw the same result with or without the code above it wasn't true, we'd been pulling the same container both times.

So without the above code added it just doesn't say anything, with that code added we get the DD rum error originally posted, with either v3 or v5

**Comment by camilaibs** [comment]
@bruno-watt-wcq thanks for the quick reply 🙂. Could you please check how the script appears in the browser? Would like to know if the configs passed to the init script are replaced correctly or if they are empty.

**Comment by bruno-watt-wcq** [comment]
Ur welcome @camilaibs . Yep I can confirm the values are being dropped in there correctly.

Because im seeing the 'number between 1 and 100' error for the sample rate I thought I would point out that those values are strings. Here's a sanitised view:

```
       DD_RUM.onReady(function () {
          DD_RUM.init({
            clientToken: 'my-very-special-token-not-this',
            applicationId:
              'ideee',
            site: 'datadoghq.eu',
            service: 'backstage',
            env: 'devops-green',
            sampleRate:
              '100',
            sessionReplaySampleRate:
              '1',
            trackInteractions: true,
          });
```

**Comment by freben** [comment]
Alright, and the quotes are there in the original code too right? Does it work if you remove those?

```html
          sampleRate:
            '<%= config.getOptionalNumber("app.datadogRum.sessionSampleRate") || 100 %>',
```

**Comment by danherd** [comment]
Please see [my comment on the recent docs PR](https://github.com/backstage/backstage/pull/25411#pullrequestreview-2278615828)
